src/Advisor/Advisor.Test/ScenarioTests/AdvisorTestRunner.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections.Generic;
+using Microsoft.Azure.Commands.TestFx;
+using Microsoft.Azure.Internal.Subscriptions;
+using Microsoft.Azure.Management.Advisor;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Commands.Advisor.Test.ScenarioTests
+{
+    public class AdvisorTestRunner
+    {
+        protected readonly ITestRunner TestRunner;
+
+        protected AdvisorTestRunner(ITestOutputHelper output)
+        {
+            TestRunner = TestManager.CreateInstance (output)
+                .WithNewPsScriptFilename ($"{GetType().Name}.ps1")
+                .WithProjectSubfolderForTests ("ScenarioTests")
+                .WithCommonPsScripts (new[]
+                {
+                    @"Common.ps1"
+                })
+                .WithNewRmModules (helper => new[]
+                {
+                    helper.RMProfileModule,
+                    helper.GetRMModulePath("Az.Advisor.psd1")
+                })
+                .WithNewRecordMatcherArguments (
+                    userAgentsToIgnore: new Dictionary<string, string>(),
+                    resourceProviders: new Dictionary<string, string>
+                    {
+                        {"Microsoft.Resources", null},
+                        {"Microsoft.Features", null},
+                        {"Microsoft.Authorization", null}
+                    }
+                ).WithManagementClients(
```

Comment:
WithManagementClients is not necessary as both GetResourceGraphClient and GetSubscriptionClient instantiate the object like context.GetServiceClient<TClient>